src/config.rs

```diff
@@ -293,6 +293,20 @@ pub trait Clock {
     /// it represents a sequence that can attribute a logical ordering
     /// to actions on the database.
     fn now(&self) -> i64;
+
+    /// SlateDB enforces monotonic clocks, even across restarts, which can result in situations where
+    /// the persisted time (from the manifest or WAL) is ahead of the current time on a machine.
+    /// In such cases, this method calculates how long the system should sleep to allow the clock
+    /// to synchronize with the last persisted time.
+    ///
+    /// # Parameters
+    ///
+    /// - `last_tick`: The last observed persisted clock tick (in milliseconds).
+    ///
+    /// # Returns
+    ///
+    /// A `Duration` representing the amount of system time to sleep to compensate for clock skew.
+    fn clock_sync_duration(&self, last_tick: i64) -> Duration;
```

Comment:
NOTE: this is technically a public API creep, and we could instead have the default behavior just do what the system clock does but the risk is that with non-system time clocks this behavior might result in a sleep that is very, very long.